dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -175,6 +178,73 @@ public WorkflowDAO(SessionFactory factory) {
+        Root<Workflow> entryRoot = q.from(Workflow.class);
+
+        List<Predicate> predicates = new ArrayList<>();
+        predicates.add(cb.isTrue(entryRoot.get("isPublished")));
+        Predicate predicate = cb.isTrue(entryRoot.get("isPublished"));
+
+        predicate = andLike(cb, predicate, entryRoot.get("organization"), Optional.ofNullable(organization));
+        predicate = andLike(cb, predicate, entryRoot.get("repository"), Optional.ofNullable(name));
+        predicate = andLike(cb, predicate, entryRoot.get("workflowName"), Optional.ofNullable(toolname));
+        predicate = andLike(cb, predicate, entryRoot.get("description"), Optional.ofNullable(description));
+        predicate = andLike(cb, predicate, entryRoot.get("author"), Optional.ofNullable(author));
+        predicate = andEqualDescriptorType(cb, predicate, entryRoot.get("descriptorType"), Optional.ofNullable(descriptorType));
+
+        if (registry != null) {
+            predicate = cb.and(predicate, cb.equal(entryRoot.get("sourceControl"), converter.convertToEntityAttribute(registry)));
+        }
+
+        if (checker != null) {
+            predicate = cb.and(predicate, cb.isTrue(entryRoot.get("isChecker")));
+        }
+
+        q.where(predicate);
+        TypedQuery<Workflow> query = currentSession().createQuery(q);
+
+        List<Workflow> workflows = query.getResultList();
+        return workflows;
+    }
+
+    private Predicate andLike(CriteriaBuilder cb, Predicate existingPredicate, Path<String> column, Optional<String> value) {
+        return value.map(val -> cb.and(existingPredicate, cb.like(column, wildcardLike(val))))
+                .orElse(existingPredicate);
+    }
+
+    private String wildcardLike(String value) {
+        return '%' + value + '%';
+    }
+
+    private Predicate andEqualDescriptorType(CriteriaBuilder cb, Predicate existingPredicate, Path<String> column, Optional<String> value) {
+        DescriptorLanguage descriptorLanguage;
+        if (value.isPresent()) {
+            String descriptorValue = value.get();
+            if ("galaxy".equalsIgnoreCase(descriptorValue)) {
+                descriptorLanguage = DescriptorLanguage.GXFORMAT2;
+            } else if (descriptorValue.equals(DescriptorLanguage.CWL.getShortName())) {
+                descriptorLanguage = DescriptorLanguage.CWL;
+            } else if (descriptorValue.equals(DescriptorLanguage.WDL.getShortName())) {
+                descriptorLanguage = DescriptorLanguage.WDL;
+            } else if (descriptorValue.equals(DescriptorLanguage.NEXTFLOW.getShortName())) {
+                descriptorLanguage = DescriptorLanguage.NEXTFLOW;
+            } else if (descriptorValue.equals(DescriptorLanguage.SERVICE.getShortName())) {
+                descriptorLanguage = DescriptorLanguage.SERVICE;
+            } else {
+                LOG.info("Cannot match descriptor type");
+                return existingPredicate;
```

Comment:
On further reflection, I don't think this is right. If the user specifies a descriptorType, you should return all the the workflows that match the descriptorType. If none match, e.g., because the descriptorType doesn't exist, then the query should return no results. Instead you're ignoring the query element.

I think you should short-circuit things and move this (or the shortened version) to the beginning of `ToolsApiServiceImpl.toolsGet` -- if there is a descriptorType query param and it doesn't resolve to a valid descriptor, just return an empty list.

In addition, you could use that DescriptorType to skip tools per my other suggestion.